network/kademlia_test.go

```diff
@@ -370,80 +390,110 @@ func binStr(a *BzzAddr) string {
 	return pot.ToBin(a.Address())[:8]
 }
 
-func TestSuggestPeerFindPeers(t *testing.T) {
-	tk := newTestKademlia(t, "00000000")
-	tk.On("00100000")
-	tk.checkSuggestPeer("<nil>", 0, false)
+//Tests peer suggestion. Depends on expectedMinBinSize implementation
+func TestSuggestPeersFindPeers(t *testing.T) {
+	base := "00000000"
+	tk := newTestKademlia(t, base)
 
+	//Add peers to bin 2 and 3 in order to be able to have depth 2
+	tk.On("00100000")
 	tk.On("00010000")
+	//No unconnected peers
 	tk.checkSuggestPeer("<nil>", 0, false)
 
-	tk.On("10000000", "10000001")
-	tk.checkSuggestPeer("<nil>", 0, false)
+	//We add some addresses that fall in bin0 and bin1
+	tk.Register("10000000", "11000000", "11100000", "11110000", "11111000")
+	tk.Register("01000000", "01100000", "01110000")
 
-	tk.On("01000000")
-	tk.Off("10000001")
-	tk.checkSuggestPeer("10000001", 0, true)
+	//Bins should fill from  most empty to least empty and shallower to deeper
+	//first suggestion should be for bin 0
+	tk.checkSuggestPeer("11111000", 0, false)
```

Comment:
Please why are we sure that exactly `11111000` is the one being suggested when the `po` is the same?